addons/mail/static/tests/mock_server/mock_models/discuss_channel_member.js

```diff
@@ -135,4 +136,41 @@ export class DiscussChannelMember extends models.ServerModel {
         const [member] = this._filter([["id", "in", ids]]);
         return ResPartner.mail_partner_format([member.partner_id])[member.partner_id];
     }
+
+    /**
+     * @param {number[]} ids
+     * @param {number} message_id
+     */
+    _set_new_message_separator(ids, message_id) {
+        const kwargs = parseModelParams(arguments, "ids", "message_id");
+        ids = kwargs.ids;
+        delete kwargs.ids;
+        message_id = kwargs.message_id;
+        const [member] = this._filter([["id", "in", ids]]);
+        if (!member) {
+            return;
+        }
+        this.env["discuss.channel.member"].write([member.id], {
+            new_message_separator: message_id,
+        });
+        const [partner, guest] = this.env["res.partner"]._get_current_persona();
+        const target = guest ?? partner;
+        const message_unread_counter = this.env["mail.message"].search_count([
+            ["res_id", "=", member.channel_id],
+            ["model", "=", "discuss.channel"],
+            ["id", ">=", message_id],
+        ]);
```

Comment:
What about isolating this into a method `_compute_message_unread`?
That way you don't repeat yourself here and in `discuss_channel.js` and this clarifies better that message_unread_counter is updated accordingly.

Reply:
Done :+1: